hedera-node/hedera-app/src/main/java/com/hedera/services/api/implementation/workflows/prehandle/PreHandleDispatcherImpl.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.api.implementation.workflows.prehandle;
+
+import com.hedera.node.app.spi.PreHandleDispatcher;
+import com.hedera.services.api.implementation.ServicesAccessor;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import java.util.Objects;
+
+public final class PreHandleDispatcherImpl implements PreHandleDispatcher {
+    private final ServicesAccessor services;
+
+    public PreHandleDispatcherImpl(ServicesAccessor services) {
+        this.services = Objects.requireNonNull(services);
+    }
+
+    @Override
+    public void dispatch(TransactionBody transactionBody) {
+        switch (transactionBody.getDataCase()) {
+                //            case FILE_CREATE ->
+                //
+                // services.fileService().preHandler().preHandleFileCreate(transactionBodyData.as());
+                //            case CRYPTO_CREATE_ACCOUNT ->
+                //
+                // services.cryptoService().preHandler().preHandleAccountCreate(transactionBodyData.as());
```

Comment:
We do have `cryptoCreate` implemented already. May be we can add it here ? Or we are thinking to add it in a future PR ?

Reply:
Good question. This code was commented out in the POC and it was unclear to me, what the plans are.